pkg/helm/fake/ociClient.go

```diff
@@ -0,0 +1,61 @@
+// Copyright 2022 the Kubeapps contributors.
+// SPDX-License-Identifier: Apache-2.0
+
+package fake
+
+import (
+	"bytes"
+	"fmt"
+	"strings"
+
+	"github.com/containerd/containerd/remotes"
+	"github.com/kubeapps/kubeapps/pkg/helm"
+	helmregistry "helm.sh/helm/v3/pkg/registry"
+)
+
+// FakeOCIClient embeds *helmregistry.Clien, but overrides the Pull method
+type FakeOCIClient struct {
+	*helmregistry.Client
+
+	ExpectedName string
+	Content      map[string]*bytes.Buffer
+	Checksum     string
+	Err          error
+}
+
+// OCIClientFactoryMock is the factory used for creating FakeOCIClients
+type OCIClientFactoryMock struct {
+	ExpectedName string
+	Content      map[string]*bytes.Buffer
+	Checksum     string
+	Err          error
+}
+
+// BuildOCIClient just passes the fileds from the factory to the FakeOCIClient
+func (cf OCIClientFactoryMock) BuildOCIClient(resolver remotes.Resolver) (helm.IOCIClient, error) {
```

Comment:
Mocked OCI client getter, using it's factory params to configure its fields (that later will be used in the `Pull` mocked function)